src/software/thunderscope/cost_vis/cost_vis.py

```diff
@@ -0,0 +1,77 @@
+import time
+import queue
+import numpy as np
+
+import pyqtgraph as pg
+from pyqtgraph.Qt import QtWidgets
+from proto.import_all_protos import *
+from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
+
+
+class CostVisualizationWidget(QtWidgets.QMainWindow):
+
+    COST_VISUALIZATION_TIMEOUT_S = 0.5
+
+    def __init__(self, buffer_size=5):
+        super(CostVisualizationWidget, self).__init__()
+        self.cost_visualization_buffer = ThreadSafeBuffer(
+            buffer_size, CostVisualization
+        )
+        self.cached_cost_vis = CostVisualization()
+        self.timeout = (
+            time.time() + CostVisualizationWidget.COST_VISUALIZATION_TIMEOUT_S
+        )
+
+        layout = pg.LayoutWidget(parent=self)
+        win = pg.GraphicsLayoutWidget(show=True)
+
+        # image settings
+        self.data = np.zeros(shape=(6, 3))
+        self.img = pg.ImageItem(self.data)
+        self.img.setBorder(pg.mkPen(color=(255, 255, 255)))
+
+        # layout settings
+        self.vb = win.addViewBox()
```

Comment:
You are probably able to call `self.vb.setRange` to manually set the range to match the field widget . More on it here: https://pyqtgraph.readthedocs.io/en/latest/api_reference/graphicsItems/viewbox.html#pyqtgraph.ViewBox.setRange
And similar to here:
https://github.com/UBC-Thunderbots/Software/blob/f2dffd6a1df824fcf6293bc3608748c960ac5c1c/src/software/thunderscope/common/proto_plotter.py#L142-L147
Note that given that the two windows may have a different size and you are locking the aspect ratio, you probably only want to set x range, or y range, but not both.
To get it working, you may also need to `disableAutoRange`, though this may not be the case.

Reply:
done